xla/service/gpu/execution_stream_assignment.cc

```diff
@@ -110,10 +117,20 @@ ExecutionStreamAssignment::ExecutionStreamAssignment(
       if (callsite.instruction()->IsAsynchronous()) {
         // Asynchronous calls will result in a new `ExecutionStreamId` being
         // dispensed for the called computations.
-        CHECK_EQ(callsite.instruction()->opcode(), HloOpcode::kAsyncStart);
-        enqueue_called_computations(callsite, next_stream_id);
+        // Special logic for explicit stream annotations.
+        std::optional<ExecutionStreamId> optional_stream_id = std::nullopt;
+        auto instruction = callsite.instruction();
+        auto it = instruction->frontend_attributes().map().find(
+            kXlaStreamAnnotationAttr);
+        if (it != instruction->frontend_attributes().map().end()) {
+          int stream_id;
+          absl::SimpleAtoi(it->second, &stream_id);
```

Comment:
Could you check the return value of this, so that the case where the frontend attribute is not a proper number, is handled?

Reply:
I added `CHECK(...)` around this method.